### PR TITLE
Use `Directory.EnumerateFiles` instead of `Directory.GetFiles` in WhichUtil.

### PR DESCRIPTION
**Pull request by TingluoHuang**
`Directory.GetFiles` will return only after all search finish.
`Directory.EnumerateFiles` will return file as they been found, so we can process them immediately.

We introduce `Which2()` in addition to `Which()`, and the usage of `Which2()` is controlled by feature flag.

After we verify the new `Which2()` is functional properly, we can switch all the code path to use `Which2()`

https://github.com/github/c2c-actions-support/issues/2902

### CODE REVIEW

File: src/Runner.Common/Constants.cs [hunk 1]
```diff
@@ -69,6 +69,8 @@ public static class Runner
             public static readonly OSPlatform Platform = OSPlatform.OSX;
 #elif OS_WINDOWS
             public static readonly OSPlatform Platform = OSPlatform.Windows;
+#else
+            public static readonly OSPlatform Platform = OSPlatform.Linux;
```
Comment (TingluoHuang) [inline src/Runner.Common/Constants.cs:73]: add a `#else` to provide default value, so VSCode won't barking on us.

File: src/Runner.Common/Constants.cs [hunk 2]
```diff
@@ -79,6 +81,8 @@ public static class Runner
             public static readonly Architecture PlatformArchitecture = Architecture.Arm;
 #elif ARM64
             public static readonly Architecture PlatformArchitecture = Architecture.Arm64;
+#else
+            public static readonly Architecture PlatformArchitecture = Architecture.X64;
```
Comment (TingluoHuang) [inline src/Runner.Common/Constants.cs:85]: same, help vscode a little bit...

File: src/Test/L0/Util/WhichUtilL0.cs [hunk 3]
```diff
@@ -212,5 +212,210 @@ public void WhichThrowsWhenSymlinkBroken()
             File.Delete(brokenSymlink);
             Environment.SetEnvironmentVariable(PathUtil.PathVariable, oldValue);
         }
+
+        [Fact]
+        [Trait("Level", "L0")]
+        [Trait("Category", "Common")]
+        public void UseWhich2FindGit()
```
Comment (TingluoHuang) [inline src/Test/L0/Util/WhichUtilL0.cs:219]: these tests are copy paste all existing tests but change `Which()` to `Which2()`

File: src/Runner.Sdk/Util/WhichUtil.cs [hunk 4]
```diff
@@ -114,6 +114,128 @@ public static string Which(string command, bool require = false, ITraceWriter tr
...
+                    }
+
+                    string[] pathExtSegments = pathExt.Split(new string[] { ";" }, StringSplitOptions.RemoveEmptyEntries);
+
+                    // if command already has an extension.
+                    if (pathExtSegments.Any(ext => command.EndsWith(ext, StringComparison.OrdinalIgnoreCase)))
+                    {
+                        try
+                        {
+                            foreach (var file in Directory.EnumerateFiles(pathSegment, command))
```
Comment (TingluoHuang) [inline src/Runner.Sdk/Util/WhichUtil.cs:177]: `GetFiles` changed to `EnumerateFiles`

File: src/Runner.Sdk/Util/WhichUtil.cs [hunk 5]
```diff
@@ -114,6 +114,128 @@ public static string Which(string command, bool require = false, ITraceWriter tr
...
+                            trace?.Verbose(ex.ToString());
+                        }
+                    }
+                    else
+                    {
+                        string searchPattern;
+                        searchPattern = StringUtil.Format($"{command}.*");
+                        try
+                        {
+                            foreach (var file in Directory.EnumerateFiles(pathSegment, searchPattern))
```
Comment (TingluoHuang) [inline src/Runner.Sdk/Util/WhichUtil.cs:198]: `GetFiles` changed to `EnumerateFiles`

File: src/Runner.Sdk/Util/WhichUtil.cs [hunk 6]
```diff
@@ -114,6 +114,128 @@ public static string Which(string command, bool require = false, ITraceWriter tr
...
+                        catch (UnauthorizedAccessException ex)
+                        {
+                            trace?.Info("Ignore UnauthorizedAccess exception during Which.");
+                            trace?.Verbose(ex.ToString());
+                        }
+                    }
+#else
+                    try
+                    {
+                        foreach (var file in Directory.EnumerateFiles(pathSegment, command))
```
Comment (TingluoHuang) [inline src/Runner.Sdk/Util/WhichUtil.cs:221]: `GetFiles` changed to `EnumerateFiles`

**Review by pje — APPROVED** [review]
Left a minor comment. Overall +1! To me—someone who's only just literate in C#—it makes sense.

**Pull request merged**